aws_lambda_powertools/utilities/idempotency/exceptions.py

```diff
@@ -83,3 +83,21 @@ class IdempotencyNoSerializationModelError(BaseError):
     """
     No model was supplied to the serializer
     """
+
+
+class IdempotencyRedisClientConfigError(BaseError):
+    """
+    The Redis connection passed in has unsupported config
+    """
+
+
+class IdempotencyRedisConnectionError(BaseError):
+    """
+    Redis connection error
+    """
+
+
+class IdempotencyOrphanRecordError(BaseError):
```

Comment:
Since this is specific to Redis, can we rename it?

```suggestion
class IdempotencyRedisOrphanRecordError(BaseError):
```

Reply:
Note: in general I'm not a fan of having exceptions that are specific to a persistence layer. I would be happier with generic names as:

- IdempotencyPersistenceConfigError
- IdempotencyPersistenceConnectionError
- IdempotencyPersistenceConsistencyError

One of the reasons is that we might add additional persistence layers in the future, and this file will only continue to grow bigger (+ cognitive load). WDYT?